pkg/service/spec.go

```diff
@@ -12,10 +12,19 @@ type processSpec struct {
 	ImageData []byte
 	// Params hold the key-value pairs for the processing job and tells the manipulator what to do with the image
 	Params map[string]string
+	// Format to target Extension
+	TargetExtension string
 	// Formats have the information of accepted formats, whether darkroom can return the image using webp or not
 	formats []string
 }
 
+const (
```

Comment:
I suggest creating a new type, say `Format`
```go
type Format int

const (
	FormatJPG = iota
	FormatPNG
	FormatWebP
)
```

ps: Can be unexported, if you'd like

Reply:
For this case we need to string value

Reply:
For this case we need to string value

Reply:
Why do we need to specifically keep it a string?
If it is because of extension being a string, then again we should be parsing extension from string as it may be in different formats, say it is capitalised, etc.

Reply:
Part of it yes, our internal darkroom will get the format as a string from the file name, tokenizing to iota will take more steps compare to make the format case insensitive. 

Another part is that this param will directly interact with [our encoder](https://github.com/gojek/darkroom/blob/e276ae0d74a17e5e2d758505b9bf50d92633086a/pkg/processor/native/processor.go#L109) which accepts the string format (`.jpeg`, `.jpg` and .`png`), converting this to iota will result in more changes required.